specs/AddHostObjectToScriptWithOrigins.md

```diff
@@ -0,0 +1,188 @@
+When script in the iframe tries to access a host object the origin of the iframe will be
+checked against the provided list. The Frame object has a Name property with the value of the
+name attribute from the iframe html tag declaring it. The NameChanged event is raised when
+corresponding iframe changes its window.name property.
+
+The host object will be removed during iframe deletion automatically.
+
+
+# Examples
+## FrameCreated/FrameDeleted and AddHostObjectToScriptWithOrigins
+
+## .NET, WinRT
+```c#
+void SubscribeToFrameCreated()
+{
+    webView.CoreWebView2.FrameCreated += delegate (object sender, CoreWebView2FrameCreatedEventArgs args)
+    {
+        try
+        {
+            List<string> origins = new List<string>(new string[] { "https://appassets.example" });
+            args.Frame.AddHostObjectToScriptWithOrigins("bridge", new BridgeAddRemoteObject(), origins, 1);
+        }
+        catch (NotSupportedException exception)
+        {
+            MessageBox.Show("Frame.AddHostObjectToScriptWithOrigins failed: " + exception.Message);
+        }
+    };
+}
+```
+
+## Win32 C++
+```cpp
+void SampleClass::SampleMethod()
+{
+    // Register a handler for the FrameCreated event.
+    // This handler can be used to add host objects to the created iframe.
+    CHECK_FAILURE(m_webView->add_FrameCreated(
+        Callback<ICoreWebView2FrameCreatedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2FrameCreatedEventArgs* args) -> HRESULT
+    {
+        wil::com_ptr<ICoreWebView2Frame> webviewFrame;
+        CHECK_FAILURE(args->get_Frame(&webviewFrame));
+
+        // Subscribe to frame deleted event
+        webviewFrame->add_FrameDeleted(
+        Callback<ICoreWebView2FrameDeletedEventHandler>(
+            [this](ICoreWebView2* sender) -> HRESULT {
+            /*Cleanup on frame deletion*/
+        }).Get(), &m_frameDeletedToken));
+
+        // Wrap host object
+        VARIANT remoteObjectAsVariant = {};
+        m_hostObject.query_to<IDispatch>(&remoteObjectAsVariant.pdispVal);
+        remoteObjectAsVariant.vt = VT_DISPATCH;
+
+        // Create list of origins which will be checked.
+        // iframe will have access to host object only if its origin belongs
+        // to this list.
+        int originsCount = 1;
+        LPCWSTR* originsList =
+            reinterpret_cast<LPCWSTR*>(CoTaskMemAlloc(sizeof(LPCWSTR) * originsCount));
+        originsList[0] = L"https://appassets.example";
+        // Add host object to a script for iframe access
+        CHECK_FAILURE(
+            webviewFrame->AddHostObjectToScriptWithOrigins(
+                L"sample", &remoteObjectAsVariant, originsList, originsCount));
+
+        remoteObjectAsVariant.pdispVal->Release();
+        CoTaskMemFree(originsList);
+
+        return S_OK;
+    }).Get(), &m_frameCreatedToken));
+}
+```
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## IDL
```

Comment:
You have the IDL API definition which is great, but you are missing the C# or IDL3 API definition. 

If you have this building in build-apiwriter.ps1 you can copy the relevant parts out of third_party\edge_webview2\win\webview2_api_writer\dotNetAPIWrapper\winrt_wrapper\Microsoft.Web.WebView2.Core.idl